tests/integration/vxlan_test.py

```diff
@@ -0,0 +1,118 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+import time
+
+import pytest
+
+import libnmstate
+from libnmstate.error import NmstateVerificationError
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceState
+from libnmstate.schema import InterfaceType
+
+from .testlib import assertlib
+from .testlib.vxlan import apply_state
+from .testlib.vxlan import vxlan_down
+from .testlib.vxlan import vxlan_interfaces
+from .testlib.vxlan import vxlan_up
+from .testlib.vxlan import vxlans_absent
+from .testlib.vxlan import vxlans_up
```

Comment:
This shouts "I am an object" :)
Not a must, but you should consider it.

Reply:
Have refactor it with Vxlan class and proper methods.